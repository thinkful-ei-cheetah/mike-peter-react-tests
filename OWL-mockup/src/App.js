import React from 'react';

import PList from './participant-comp/participantList';
import Stage from './stage-comp/stage';
import Chat from './chat-comp/chatbar'



function App(props) {

const pList = <PList people={props.store.participants} />
const stage = <Stage actors={props.store.participants}/>
const chat = <Chat />




  return (
    <main className='App'>
      <header className="App-header">
        <h1>OWL markup</h1>
       

    </header>
      {pList} 
      {stage}
      {chat}
    </main>
  );
}

export default App;