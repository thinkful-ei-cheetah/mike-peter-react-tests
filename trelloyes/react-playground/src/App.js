import React from 'react';
import List from './List';


function App(props) {

const list = props.store.lists.map( list => 

  <List id={list.id} header={list.header} cardIds={list.cardIds} store={props.store} />

//   <section id="list.id" className="List">
//    <header className="List-header">
//        <h2>{list.header}</h2>
//   </header>

//   <div className="List-cards">
//   </div>
      
// </section>
  );


  return (
    <main className='App'>
      <header className="App-header">
        <h1>Trelloyes!</h1>
        
      </header>
      {list}
    </main>
  );
}

export default App;

