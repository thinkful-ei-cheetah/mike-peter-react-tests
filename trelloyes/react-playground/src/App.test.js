import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import STORE from './store.js';
import renderer from 'react-test-renderer';

//this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  //render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<App store={STORE} />, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<App store={STORE} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});

it('renders the UI as expected with no unreads', () => {
  const tree = renderer
    .create(<App store={STORE} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});