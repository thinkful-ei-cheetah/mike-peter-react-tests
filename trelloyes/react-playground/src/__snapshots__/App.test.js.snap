// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Trelloyes!
    </h1>
  </header>
  <section
    className="List"
    id="1"
  >
    <header
      className="List-header"
    >
      <h2>
        First list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="2"
  >
    <header
      className="List-header"
    >
      <h2>
        Second list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="3"
  >
    <header
      className="List-header"
    >
      <h2>
        Third list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="4"
  >
    <header
      className="List-header"
    >
      <h2>
        Fourth list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
</main>
`;

exports[`renders the UI as expected with no unreads 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Trelloyes!
    </h1>
  </header>
  <section
    className="List"
    id="1"
  >
    <header
      className="List-header"
    >
      <h2>
        First list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="2"
  >
    <header
      className="List-header"
    >
      <h2>
        Second list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="3"
  >
    <header
      className="List-header"
    >
      <h2>
        Third list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            First card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Second card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Third card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fourth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Fifth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Sixth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Seventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eighth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Ninth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Tenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Eleventh card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="List"
    id="4"
  >
    <header
      className="List-header"
    >
      <h2>
        Fourth list
      </h2>
    </header>
    <div
      className="List-cards"
    >
      <div>
        <div
          className="card"
        >
          <h3>
            Twelth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
        <div
          className="card"
        >
          <h3>
            Thirteenth card
          </h3>
          <p>
            lorem ipsum
          </p>
        </div>
      </div>
    </div>
  </section>
</main>
`;
